### PR TITLE
[build-utils] add pnpm7 to path when lockfile v5.4 is detected

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
### Related Issues

Adds support for `pnpm@7` using a similar lock file detection and path setting method that `npm@7` uses

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

**Review by styfle — COMMENTED** [review]
Looks good. Lets put this on hold until the build image ships

File: packages/build-utils/src/fs/run-user-scripts.ts [hunk 1]
```diff
@@ -442,6 +442,16 @@ export function getEnvForPackageManager({
...
       console.log('Detected `package-lock.json` generated by npm 7...');
     }
+  } else if (cliType === 'pnpm') {
+    if (
+      typeof lockfileVersion === 'number' &&
+      lockfileVersion === 5.4 &&
+      (nodeVersion?.major || 0) > 12
+    ) {
+      // Ensure that pnpm 7 is at the beginning of the `$PATH`
+      newEnv.PATH = `/pnpm7/pnpm:${env.PATH}`;
```
Comment (styfle) [inline packages/build-utils/src/fs/run-user-scripts.ts:452]:
```suggestion
      newEnv.PATH = `/pnpm7:${env.PATH}`;
```

Wouldn't it be the directory only?

Might be good to add the e2e test to confirm like you didn't for the other pnpm tests.

Comment (Ethan-Arrowood) [reply packages/build-utils/src/fs/run-user-scripts.ts:452]: Oh maybe - the executable itself is at `pnpm7/pnpm` but i guess for path you just need the directory not the executable. I will add an integration test now

Comment (Ethan-Arrowood) [reply packages/build-utils/src/fs/run-user-scripts.ts:452]: this doesn't work if its not the full executable path

File: packages/build-utils/test/fixtures/28-pnpm-7/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "22-pnpm",
```
Comment (styfle) [inline packages/build-utils/test/fixtures/28-pnpm-7/package.json]: Lets remove everything except `scripts` and `dependencies`. And then also add `private: true`

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**